### PR TITLE
ROX-20492: Delete autogenerated integrations if feature disabled.

### PR DESCRIPTION
**Pull request by dcaravel**
## Description

Similar to `sourced` autogenerated integrations, with this change normal autogenerated integrations will be deleted on Central startup if disabled via `ROX_DISABLE_AUTOGENERATED_REGISTRIES`=`true`.

## Checklist
- [x] Investigated and inspected CI test results
- [X] Unit test and regression tests added
- [X] Evaluated and added CHANGELOG entry if required
- [X] Determined and documented upgrade steps
- [X] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed
Unit tests have been added and installed ACS on a fresh OCP cluster, observed that default autogenerated integrations were created.

```
imageintegration/datastore: 2023/11/03 20:24:33.785025 singleton.go:53: Info: [STARTUP] Starting deletion of autogenerated image integrations (autogen disabled [false], 'sourced' autogen disabled [true])
imageintegration/datastore: 2023/11/03 20:24:33.785191 singleton.go:73: Info: Completed deletion of image integrations
pkg/registries/docker: 2023/11/03 20:24:34.528801 docker.go:128: Debug: created integration "Public DockerHub" with repo list disabled
pkg/registries/docker: 2023/11/03 20:24:34.530069 docker.go:128: Debug: created integration "Public Kubernetes GCR (deprecated)" with repo list disabled
pkg/registries/docker: 2023/11/03 20:24:34.530590 docker.go:128: Debug: created integration "Public Kubernetes Registry" with repo list disabled
pkg/registries/docker: 2023/11/03 20:24:34.531436 docker.go:128: Debug: created integration "Public GCR" with repo list disabled
pkg/registries/docker: 2023/11/03 20:24:34.532096 docker.go:128: Debug: created integration "Public GKE GCR" with repo list disabled
pkg/registries/docker: 2023/11/03 20:24:34.532725 docker.go:128: Debug: created integration "Public Quay.io" with repo list disabled
pkg/registries/docker: 2023/11/03 20:24:34.533356 docker.go:128: Debug: created integration "Public Microsoft Container Registry" with repo list disabled
pkg/registries/docker: 2023/11/03 20:24:34.533928 docker.go:128: Debug: created integration "registry.access.redhat.com" with repo list disabled
pkg/registries/docker: 2023/11/03 20:24:35.630189 docker.go:126: Debug: created integration "Autogenerated https://quay.io for cluster remote" with repo list enabled
pkg/registries/docker: 2023/11/03 20:24:35.665858 docker.go:126: Debug: created integration "Autogenerated https://cloud.openshift.com for cluster remote" with repo list enabled
pkg/registries/docker: 2023/11/03 20:24:35.979179 docker.go:126: Debug: created integration "Autogenerated https://registry.connect.redhat.com for cluster remote" with repo list enabled
pkg/registries/docker: 2023/11/03 20:24:36.161001 docker.go:126: Debug: created integration "Autogenerated https://registry.redhat.io for cluster remote" with repo list enabled
```

Then executed `k set env deploy/central -n stackrox ROX_DISABLE_AUTOGENERATED_REGISTRIES=true`

And observed via logs the autogenerated integrations were deleted and stayed deleted.

```
imageintegration/datastore: 2023/11/03 20:27:40.041249 singleton.go:53: Info: [STARTUP] Starting deletion of autogenerated image integrations (autogen disabled [true], 'sourced' autogen disabled [true])
imageintegration/datastore: 2023/11/03 20:27:40.049408 singleton.go:70: Info: Successfully deleted 4 out of 4 image integration(s)
imageintegration/datastore: 2023/11/03 20:27:40.049485 singleton.go:73: Info: Completed deletion of image integrations
pkg/registries/docker: 2023/11/03 20:27:40.791717 docker.go:128: Debug: created integration "Public DockerHub" with repo list disabled
pkg/registries/docker: 2023/11/03 20:27:40.794049 docker.go:128: Debug: created integration "Public Kubernetes GCR (deprecated)" with repo list disabled
pkg/registries/docker: 2023/11/03 20:27:40.795049 docker.go:128: Debug: created integration "Public Kubernetes Registry" with repo list disabled
pkg/registries/docker: 2023/11/03 20:27:40.796086 docker.go:128: Debug: created integration "Public GCR" with repo list disabled
pkg/registries/docker: 2023/11/03 20:27:40.796820 docker.go:128: Debug: created integration "Public GKE GCR" with repo list disabled
pkg/registries/docker: 2023/11/03 20:27:40.797463 docker.go:128: Debug: created integration "Public Quay.io" with repo list disabled
pkg/registries/docker: 2023/11/03 20:27:40.798432 docker.go:128: Debug: created integration "Public Microsoft Container Registry" with repo list disabled
pkg/registries/docker: 2023/11/03 20:27:40.799216 docker.go:128: Debug: created integration "registry.access.redhat.com" with repo list disabled
```

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: central/imageintegration/datastore/singleton.go [hunk 1]
```diff
@@ -20,40 +21,56 @@ var (
...
 			}
 		}
-		if attempted > 0 {
-			log.Infof("Successfully deleted %d out of %d image integration(s)", deleted, attempted)
-		}
-
-		log.Info("Completed deletion of 'sourced' image integrations")
 	}
+	if attempted > 0 {
+		log.Infof("Successfully deleted %d out of %d image integration(s)", deleted, attempted)
```
Comment (RTann) [inline central/imageintegration/datastore/singleton.go:70]: can we have a log for when none are deleted?

**Review by RTann — COMMENTED** [review]
just curious since @connorgorman is pretty familiar with autogenerated registries if he sees anything wrong with deleting them upon startup when they are disabled

**Review by connorgorman — APPROVED** [review]

File: central/imageintegration/datastore/singleton_test.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/features"
+	"go.uber.org/mock/gomock"
+)
+
+// mustUpdateSourcedAutogenFeature will attempt to update the the sourced autogenerated
+// feature to the desired value, if unable to update the feature value will skip the
+// the test.
+func mustUpdateSourcedAutogenFeature(t *testing.T, desired bool) {
+	t.Setenv(features.SourcedAutogeneratedIntegrations.EnvVar(), fmt.Sprintf("%t", desired))
```
Comment (RTann) [inline central/imageintegration/datastore/singleton_test.go:19]: nit: `strconv.FormatBool` is preferred

File: central/imageintegration/datastore/singleton_test.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+			[]string{},
+			true,
+			true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			mustUpdateSourcedAutogenFeature(t, tc.sourcedAutogenEnabled)
+			t.Setenv(env.AutogeneratedRegistriesDisabled.EnvVar(), fmt.Sprintf("%t", !tc.autogenEnabled))
```
Comment (RTann) [inline central/imageintegration/datastore/singleton_test.go:72]: nit: `strconv.FormatBool`

**Review by RTann — APPROVED** [review]
LGTM just one nit twice

**Pull request merged**